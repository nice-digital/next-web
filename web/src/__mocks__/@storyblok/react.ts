export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
