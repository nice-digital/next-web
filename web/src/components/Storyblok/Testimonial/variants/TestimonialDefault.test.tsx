/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";
import {
	TestimonialDefault,
	TestimonialDefaultProps,
} from "./TestimonialDefault";

const mockTestimonialDefault: TestimonialDefaultProps = {
	blok: {
		image: {
			id: 123456789,
			name: "Kittens mate.",
			filename: "https://placekitten.com/408/287",
			alt: null,
			fieldtype: "asset",
			title: null,
			focus: null,
		},
		quoteName: "Test Name",
		quoteRole: "Test Role",
		quoteText: "Test Quote Text",
		variant: "transparent",
		_uid: "",
		component: "testimonialDefault",
	},
};

const mockTestimonialDefaultTransparent: TestimonialDefaultProps = {
	blok: {
		...mockTestimonialDefault.blok,
		variant: "transparent",
	},
};

describe("Storyblok testimonial Default component", () => {
	it("should match snapshot for default variant", () => {
		const { container } = render(
			<TestimonialDefault {...mockTestimonialDefault} />
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot for the default transparent variant", () => {
		const { container } = render(
			<TestimonialDefault {...mockTestimonialDefaultTransparent} />
		);
		expect(container).toMatchSnapshot();
	});
});
