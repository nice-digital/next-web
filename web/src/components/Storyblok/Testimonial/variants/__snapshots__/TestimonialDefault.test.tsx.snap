// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyblok testimonial Default component should match snapshot for default variant 1`] = `
<div>
  <div
    class="testimonial testimonial--transparent"
    data-component="testimonial--transparent"
    data-testid="testimonial-transparent"
  >
    <div
      class="testimonialMainContent"
    >
      <div
        class="testimonial__mainImageContainer"
      >
        <picture>
          <source
            srcset="https://placekitten.com/408/287/m/708x0/filters:format%28avif%29:quality%2880%29"
            type="image/avif"
          />
          <source
            srcset="https://placekitten.com/408/287/m/708x0/filters:format%28webp%29:quality%2880%29"
            type="image/webp"
          />
          <img
            alt="some alt text"
            class="testimonialMainImage"
            src="https://placekitten.com/408/287/m/708x0/filters:format%28jpeg%29:quality%2880%29"
          />
        </picture>
      </div>
      <div
        class="testimonialContent"
      >
        <p
          class="testimonial__quote"
        >
          Test Quote Text
        </p>
        <div
          class="testimonial__person"
        >
          <picture>
            <source
              srcset="https://placekitten.com/408/287/m/708x0/filters:format%28avif%29:quality%2880%29"
              type="image/avif"
            />
            <source
              srcset="https://placekitten.com/408/287/m/708x0/filters:format%28webp%29:quality%2880%29"
              type="image/webp"
            />
            <img
              alt="some alt text"
              class="testimonial__mobileImage"
              src="https://placekitten.com/408/287/m/708x0/filters:format%28jpeg%29:quality%2880%29"
            />
          </picture>
          <div
            class="testimonial__details"
          >
            <p
              class="testimonial__name"
            >
              Test Name
            </p>
            <p
              class="testimonial__job"
            >
              Test Role
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyblok testimonial Default component should match snapshot for the default transparent variant 1`] = `
<div>
  <div
    class="testimonial testimonial--transparent"
    data-component="testimonial--transparent"
    data-testid="testimonial-transparent"
  >
    <div
      class="testimonialMainContent"
    >
      <div
        class="testimonial__mainImageContainer"
      >
        <picture>
          <source
            srcset="https://placekitten.com/408/287/m/708x0/filters:format%28avif%29:quality%2880%29"
            type="image/avif"
          />
          <source
            srcset="https://placekitten.com/408/287/m/708x0/filters:format%28webp%29:quality%2880%29"
            type="image/webp"
          />
          <img
            alt="some alt text"
            class="testimonialMainImage"
            src="https://placekitten.com/408/287/m/708x0/filters:format%28jpeg%29:quality%2880%29"
          />
        </picture>
      </div>
      <div
        class="testimonialContent"
      >
        <p
          class="testimonial__quote"
        >
          Test Quote Text
        </p>
        <div
          class="testimonial__person"
        >
          <picture>
            <source
              srcset="https://placekitten.com/408/287/m/708x0/filters:format%28avif%29:quality%2880%29"
              type="image/avif"
            />
            <source
              srcset="https://placekitten.com/408/287/m/708x0/filters:format%28webp%29:quality%2880%29"
              type="image/webp"
            />
            <img
              alt="some alt text"
              class="testimonial__mobileImage"
              src="https://placekitten.com/408/287/m/708x0/filters:format%28jpeg%29:quality%2880%29"
            />
          </picture>
          <div
            class="testimonial__details"
          >
            <p
              class="testimonial__name"
            >
              Test Name
            </p>
            <p
              class="testimonial__job"
            >
              Test Role
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
